addons/crm/models/crm_lead.py

```diff
@@ -283,6 +283,17 @@ def _onchange_next_activity_id(self):
     # ----------------------------------------
 
     @api.model
+    def name_create(self, name):
+        res = super(Lead, self).name_create(name)
+        
+        # update the probability of the lead if the stage is set to update it automatically
+        lead_obj = self.env['crm.lead'].browse(res[0])
+        stage = lead_obj.stage_id
+        if stage.on_change:
+            lead_obj.update( {'probability': stage.probability} )
```

Comment:
To double check, but all this can work with:
``` python
res = super(Lead, self).name_create(name)
self.browse(res[0])._onchange_stage_id()
return res
```
If this works as expected, you can merge.